File: backend/app/schemas/lesson.py
from pydantic import BaseModel, ConfigDict

class LessonBase(BaseModel):
    title: str
    content_type: str
    content_data: str
    order: int

class LessonCreate(LessonBase):
    pass

class Lesson(LessonBase):
    id: int
    course_id: int

    model_config = ConfigDict(from_attributes=True)

File: frontend/lib/domain/entities/lesson.dart
// frontend/lib/domain/entities/lesson.dart
import 'package:equatable/equatable.dart';

class Lesson extends Equatable {
  final int id;
  final int courseId;
  final String title;
  final String contentType;
  final String contentData;
  final int order;

  const Lesson({
    required this.id,
    required this.courseId,
    required this.title,
    required this.contentType,
    required this.contentData,
    required this.order,
  });

  @override
  List<Object?> get props => [id, courseId, title, contentType, contentData, order];
}
